apis/workstations/v1alpha1/config_identity.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// WorkstationConfigIdentity defines the resource reference to WorkstationConfig, which "External" field
+// holds the GCP identifier for the KRM object.
+type WorkstationConfigIdentity struct {
+	parent *WorkstationConfigParent
+	id     string
+}
+
+func (i *WorkstationConfigIdentity) String() string {
+	return i.parent.String() + "/workstationConfigs/" + i.id
+}
+
+func (i *WorkstationConfigIdentity) ID() string {
+	return i.id
+}
+
+func (i *WorkstationConfigIdentity) Parent() *WorkstationConfigParent {
+	return i.parent
+}
+
+type WorkstationConfigParent struct {
+	ProjectID string
+	Location  string
+	Cluster   string
+}
+
+func (p *WorkstationConfigParent) String() string {
+	return "projects/" + p.ProjectID + "/locations/" + p.Location + "/workstationClusters/" + p.Cluster
+}
+
+// New builds a ConfigIdentity from the Config Connector WorkstationConfig object.
+func NewWorkstationConfigIdentity(ctx context.Context, reader client.Reader, obj *WorkstationConfig) (*WorkstationConfigIdentity, error) {
+	// Get Parent
+	clusterRef := obj.Spec.Parent
+	if clusterRef == nil {
+		return nil, fmt.Errorf("no parent cluster")
+	}
+	clusterExternal, err := clusterRef.NormalizedExternal(ctx, reader, obj.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("cannot resolve cluster: %w", err)
+	}
+	clusterParent, cluster, err := parseWorkstationClusterExternal(clusterExternal)
```

Comment:
Personally I like it, we might end up add back the `spec.projectRef` and `spec.location` to give user a consistent experience compared to the other resources. If that's the case, what you did here would be a valuable thing and a hidden gem for those KCC lovers to explore and discover.